### PR TITLE
Wrap ternary

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request wraps the GMT function `ternary`.

**Preview** docs at https://pygmt-git-wrap-ternary-gmt.vercel.app/api/generated/pygmt.Figure.ternary.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
I'm getting a weird problem where the axis labels on the right-side (**b**) axis isn't showing up. I'm not sure what I'm missing, as this follows along with the example on the GMT page for `ternary`, and there are still gridlines for that axis appearing in the final figure.

```
fig = pygmt.Figure()
pygmt.makecpt(cmap="turbo", series=[0, 80, 10])
fig.ternary(table="@ternary.txt",
            region="0/100/0/100/0/100",
            cmap=True,
            projection="X6i",
            frame= ["bafg+lAir",
                   "cafg+lLimestone",
                   "aafg+lWater"],
            S="c0.1c")
fig.show()
```
image

**Comment by meghanrjones** [comment]
> I'm getting a weird problem where the axis labels on the right-side (**b**) axis isn't showing up. I'm not sure what I'm missing, as this follows along with the example on the GMT page for `ternary`, and there are still gridlines for that axis appearing in the final figure.
> 
> ```
> fig = pygmt.Figure()
> pygmt.makecpt(cmap="turbo", series=[0, 80, 10])
> fig.ternary(table="@ternary.txt",
>             region="0/100/0/100/0/100",
>             cmap=True,
>             projection="X6i",
>             frame= ["bafg+lAir",
>                    "cafg+lLimestone",
>                    "aafg+lWater"],
>             S="c0.1c")
> fig.show()
> ```
> 
> image

This was a bug in GMT in 6.2.0, which was fixed by https://github.com/GenericMappingTools/gmt/pull/5431.

**Comment by meghanrjones** [comment]
> > This was a bug in GMT in 6.2.0, which was fixed by [GenericMappingTools/gmt#5431](https://github.com/GenericMappingTools/gmt/pull/5431).
> 
> @meghanrjones Is there a workaround for GMT 6.2.0, or will this plotting bug be the case until GMT releases a new version and PyGMT uses it?

You could try using pygmt.config(MAP_FRAME_AXIS, "WSE"), though using that internally would prevent users from overriding it unless you found a way to check whether the setting has been configured in the session.

**Comment by willschlitzer** [comment]
> 
> You could try using pygmt.config(MAP_FRAME_AXIS, "WSE"), though using that internally would prevent users from overriding it unless you found a way to check whether the setting has been configured in the session.
> 
> Edit: To be more clear, it would need to be in a context manager for pygmt.ternary calls so that it doesn't impact other function calls.

I think calling `ternary` with a context manager will cause more confusion when none of the other plotting functions do, especially since this will presumably go back to normal in the next release of GMT (which I'm assuming will be after PyGMT v0.5.0 release). My thought is to just hold off on merging the upstream fix takes effect, as there are other methods to make `ternary` plots with Python.

**Comment by weiji14** [comment]
Since we've bumped PyGMT's minimum GMT version to 6.3.0, do you want to give this PR another go @willschlitzer?

File: pygmt/src/ternary.py
Comment (weiji14) [inline pygmt/src/ternary.py:2]:
```suggestion
ternary - Plot data on ternary diagrams.
```

File: pygmt/src/ternary.py
Comment (weiji14) [inline pygmt/src/ternary.py:14]: Seems like `tenary` uses [`-JX`](https://docs.generic-mapping-tools.org/6.3/ternary.html#j) only, so should we alias it like so?

```suggestion
    JX="projection",
```

File: pygmt/tests/test_ternary.py
Comment (weiji14) [inline pygmt/tests/test_ternary.py:2]:
```suggestion
Tests ternary.
```

File: pygmt/tests/test_ternary.py
Comment (weiji14) [inline pygmt/tests/test_ternary.py:18]: Is it possible to modify the test so that it doesn't require `makecpt`? Just want this to be a proper 'unit' test that tests the `ternary` function only.

**Review by weiji14 — COMMENTED** [review]
First round of review. Implementation look ok, but the docs could use a bit of work. Could you please add `ternary` to `doc/api/index.rst` so that we can see the preview docs? Thanks!

File: pygmt/tests/test_ternary.py
Comment (willschlitzer) [reply pygmt/tests/test_ternary.py:18]: I'll see if I can use a color pallete without `makecpt`; I've only tried it once but was unsuccessful.

Comment (willschlitzer) [reply pygmt/tests/test_ternary.py]: It's not the best graph, but I changed the test to use `cmap="red,orange,yellow,green,blue,violet",`. It doesn't apply color to all of the plotted data points, but I want to avoid an exceptionally long string for the CPT.

File: pygmt/src/ternary.py
Comment (weiji14) [inline pygmt/src/ternary.py:31]:
```suggestion
    Full option list at :gmt-docs:`ternary.html`
```

File: pygmt/tests/test_ternary.py
Comment (weiji14) [inline pygmt/tests/test_ternary.py:20]: Set an outline pen so that the white dots still show up.
```suggestion
        style="c0.1c",
        pen="thinnest",
```

File: pygmt/src/ternary.py
Comment (weiji14) [inline pygmt/src/ternary.py:29]: Probably need to make this docstring more Pythonic, but here's a first shot copied from https://github.com/GenericMappingTools/gmt/blob/6.3.0/doc/rst/source/ternary.rst
```suggestion
    Reads (*a*,\ *b*,\ *c*\ [,\ *z*]) records from *table* and plots symbols at
    those locations on a ternary diagram. If a symbol is selected and no symbol
    size given, then we will interpret the fourth column of the input data as
    symbol size. Symbols whose *size* is <= 0 are skipped. If no symbols are
    specified then the symbol code (see **style** below) must be present as
    last column in the input.  If **style** is not specified then we instead
    plot lines or polygons.
```

File: pygmt/tests/baseline/test_ternary.png.dvc
Comment (weiji14) [inline pygmt/tests/baseline/test_ternary.png.dvc:3]: Make sure to update the plot with the new pen.

Comment (willschlitzer) [reply pygmt/tests/baseline/test_ternary.png.dvc]: Updated in 33d4cb0

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:14]: I don't think `JX="projection"` is a good alias for ternary. For other modules, we usually pass `projection="X10c"`, but in ternary, we need to pass `projection="10c"`. Such inconsistency may confuse users. Can we simply use alias `JX` to `withd` instead?

Comment (willschlitzer) [reply pygmt/src/ternary.py]: updated

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:48]: Please update the docstring of `width` following the official documentation at https://docs.generic-mapping-tools.org/latest/ternary.html#j

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:26]: We should use `data`, not `table`, following the discussions in https://github.com/GenericMappingTools/pygmt/issues/1479.

Comment (willschlitzer) [reply pygmt/src/ternary.py]: done in ce6de7c

File: pygmt/src/ternary.py
Comment (willschlitzer) [reply pygmt/src/ternary.py]: I added the second line, but I think the first line doesn't make sense in the context (what would setting the width have to do with a valid linear plot?). I wanted to leave in my original line explaining how the width is set, as that is not clear from the GMT docstring.

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:12]: I don't see the `-F` option in ternary.
```suggestion
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:25]:
```suggestion
@kwargs_to_strings(R="sequence", c="sequence_comma", p="sequence")
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:21]:
```suggestion
    Y="yshift",
    c="panel",
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:28]:
```suggestion
    Reads (*a*,\ *b*,\ *c*\ [,\ *z*]) records from *data* and plots symbols at
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:32]:
```suggestion
    specified then the symbol code (see ``style`` below) must be present as
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:33]:
```suggestion
    last column in the input.  If ``style`` is not specified then we instead
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:45]:
```suggestion
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:47]:
```suggestion
        Set the width of the figure by passing a number, followed by
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:60]:
```suggestion
    {V}
    {c}
```

File: pygmt/tests/test_ternary.py
Comment (seisman) [inline pygmt/tests/test_ternary.py:16]:
```suggestion
        region=[0, 100, 0, 100, 0, 100],
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:53]: `G` is aliased to `fill`, but the docstring here says `color`. Xref https://github.com/GenericMappingTools/pygmt/issues/1617

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:56]: Need to expand the docstring of `style`, so that users understand what `style="c0.2c"` means.

File: pygmt/tests/test_ternary.py
Comment (seisman) [inline pygmt/tests/test_ternary.py:15]: Need to cache this file using the new function `load_sample_data`.

File: pygmt/src/ternary.py
Comment (willschlitzer) [reply pygmt/src/ternary.py:56]: Is there a standard as to what should be used? `pygmt.plot` has the same docstring for `style`.

File: pygmt/src/ternary.py
Comment (willschlitzer) [reply pygmt/src/ternary.py:53]: I tried using `G=["fill", "color"]` but it didn't pass the tesing/documentation build. Am I missing a step?

File: pygmt/tests/test_ternary.py
Comment (willschlitzer) [reply pygmt/tests/test_ternary.py:15]: Does this need to be cached for `load_sample_data` or is it enough to cache it for the tests?

File: pygmt/src/ternary.py
Comment (meghanrjones) [reply pygmt/src/ternary.py:56]: I think rather than adding a lot of text, we could add a link to the [symbols and markers gallery page](https://www.pygmt.org/dev/gallery/index.html#symbols-and-markers) similar to what is done for the region, frame, and projection docstring sections.

File: pygmt/src/ternary.py
Comment (seisman) [reply pygmt/src/ternary.py:53]: > I tried using `G=["fill", "color"]` but it didn't pass the tesing/documentation build. Am I missing a step?

Need the rewrite the `use_alias` decorator to support the multi-alias mechanism like `G=["fill", "color"]`.

File: pygmt/tests/test_ternary.py
Comment (seisman) [reply pygmt/tests/test_ternary.py:15]: I think it depends on if you want to add an example for ternary. If the cache data is used in an example, I think it would be better to add it to the `load_sample_data` function. Of course, it can/should be done in a separate PR. So, caching it in CI is enough for this PR.

Comment (seisman) [reply pygmt/tests/test_ternary.py:15]: Just one more thought on the test: perhaps we should use a much simpler dataset (with only a few records) in a test, so that it's easier to verify if the test is correct or not. The current test can be a good example instead.

File: pygmt/src/ternary.py
Comment (willschlitzer) [reply pygmt/src/ternary.py:53]: I think that falls outside of the scope of this pull request. I'll convert this to draft for the time being.

File: pygmt/tests/test_ternary.py
Comment (willschlitzer) [reply pygmt/tests/test_ternary.py]: I changed this to use a simple dataset

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:70]:
```suggestion
            lib.call_module("ternary", build_arg_string(kwargs, infile=infile))
```

File: pygmt/tests/test_ternary.py
Comment (seisman) [inline pygmt/tests/test_ternary.py:46]:
```suggestion
```

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:51]:
```suggestion
    region : str or list
        [*amin*, *amax*, *bmin*, *bmax*, *cmin*, *cmax*].
        Give the min and max limits for each of the three axis **a**, **b**, and **c**.
```

**Comment by meghanrjones** [comment]
Hi @willschlitzer, I think this PR is good to merge after two small in-line suggestions (one for a line length issue to satisfy pylint and one to add a reference to plotting styles). Would you be open to moving forward with a merge rather than closing the PR?

**Comment by willschlitzer** [comment]
@meghanrjones Not sure how I managed this (currently on a work computer), but didn't mean to close it! Thanks

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:71]:
```suggestion
            lib.call_module(
                module="ternary", 
                args=build_arg_string(kwargs, infile=infile),
            )
```

File: pygmt/src/ternary.py
Comment (seisman) [reply pygmt/src/ternary.py:55]: I think we can ignore it now and address the issue in https://github.com/GenericMappingTools/pygmt/issues/1617 instead.

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:58]:
```suggestion
    style : str
    	*symbol*\[\ *size*].
    	Plot individual symbols in a ternary diagram.
```

**Comment by seisman** [comment]
@willschlitzer I think this PR is almost ready for merge. I'm wondering if you have some time to finalize this PR recently.

File: pygmt/src/ternary.py
Comment (seisman) [inline pygmt/src/ternary.py:53]:
```suggestion
        Give the min and max limits for each of the three axes **a**, **b**, 
        and **c**.
```

**Comment by seisman** [comment]
The tests fail, likely due to the outdated baseline image. Could you please regenerate the baseline image using GMT 6.4.0 again?

**Review by seisman — APPROVED** [review]

**Pull request merged**
